src/cluster-legacy.c

```diff
@@ -5901,87 +5655,236 @@ sds genClusterInfoString(void) {
-"    Assign slots which are between <start-slot> and <end-slot> to current node.",
-"BUMPEPOCH",
-"    Advance the cluster config epoch.",
-"COUNT-FAILURE-REPORTS <node-id>",
-"    Return number of failure reports for <node-id>.",
-"COUNTKEYSINSLOT <slot>",
-"    Return the number of keys in <slot>.",
-"DELSLOTS <slot> [<slot> ...]",
-"    Delete slots information from current node.",
-"DELSLOTSRANGE <start slot> <end slot> [<start slot> <end slot> ...]",
-"    Delete slots information which are between <start-slot> and <end-slot> from current node.",
-"FAILOVER [FORCE|TAKEOVER]",
-"    Promote current replica node to being a master.",
-"FORGET <node-id>",
-"    Remove a node from the cluster.",
-"GETKEYSINSLOT <slot> <count>",
-"    Return key names stored by current node in a slot.",
-"FLUSHSLOTS",
-"    Delete current node own slots information.",
-"INFO",
-"    Return information about the cluster.",
-"KEYSLOT <key>",
-"    Return the hash slot for <key>.",
-"MEET <ip> <port> [<bus-port>]",
-"    Connect nodes into a working cluster.",
-"MYID",
-"    Return the node id.",
-"MYSHARDID",
-"    Return the node's shard id.",
-"NODES",
-"    Return cluster configuration seen by node. Output format:",
-"    <id> <ip:port@bus-port[,hostname]> <flags> <master> <pings> <pongs> <epoch> <link> <slot> ...",
-"REPLICATE <node-id>",
-"    Configure current node as replica to <node-id>.",
-"RESET [HARD|SOFT]",
-"    Reset current node (default: soft).",
-"SET-CONFIG-EPOCH <epoch>",
-"    Set config epoch of current node.",
-"SETSLOT <slot> (IMPORTING <node-id>|MIGRATING <node-id>|STABLE|NODE <node-id>)",
-"    Set slot state.",
-"REPLICAS <node-id>",
-"    Return <node-id> replicas.",
-"SAVECONFIG",
-"    Force saving cluster configuration on disk.",
-"SLOTS",
-"    Return information about slots range mappings. Each range is made of:",
-"    start, end, master and replicas IP addresses, ports and ids",
-"SHARDS",
-"    Return information about slot range mappings and the nodes associated with them.",
-"LINKS",
-"    Return information about all network links between this node and its peers.",
-"    Output format is an array where each array element is a map containing attributes of a link",
-NULL
-        };
-        addReplyHelp(c, help);
-    } else if (!strcasecmp(c->argv[1]->ptr,"meet") && (c->argc == 4 || c->argc == 5)) {
+    /* Terminate the link based on the direction or all. */
+    if (!strcasecmp(c->argv[3]->ptr, "from")) {
+        freeClusterLink(n->inbound_link);
+    } else if (!strcasecmp(c->argv[3]->ptr, "to")) {
+        freeClusterLink(n->link);
+    } else if (!strcasecmp(c->argv[3]->ptr, "all")) {
+        freeClusterLink(n->link);
+        freeClusterLink(n->inbound_link);
+    } else {
+        addReplyErrorFormat(c, "Unknown direction %s", (char *) c->argv[3]->ptr);
+    }
+    addReply(c, shared.ok);
+
+    return 1;
+}
+
+int clusterNodeConfirmedReachable(clusterNode  *node) {
```

Comment:
This isn't really "confirmed" reachable. This is just nodes that have been accepted into the cluster, like if a node is gossiped about this will return true even though we may never be able to reach it. So we might want to invert this to be like:

```suggestion
int clusterNodeIsPending(clusterNode *node) {
```

and just return the check? Like in flotilla this will never return, presumably, since nodes will never be added if they aren't confirmed.

Reply:
done